### ISSUE TITLE
Quarkus hot reload produces mysterious error when neo4j dependency is used

### ISSUE THREAD

**Issue by mklueh** [neo4j/neo4j#13314]
I'm facing an issue that is only occuring when Quarkus hot reload kicks in, but not on the main start of the application.

```
2023-09-29 10:35:21,028 ERROR [io.qua.dep.dev.JavaCompilationProvider] (Timer-0) Ung�ltige Servicekonfigurationsdatei, oder Ausnahme beim Erstellen des Prozessorobjekts ausgel�st: javax.annotation.processing.Processor: Provider org.neo4j.annotations.service.ServiceAnnotationProcessor could not be instantiated, line -1 in [unknown source]
Compilation Failed:
Fehler: Ung�ltige Servicekonfigurationsdatei, oder Ausnahme beim Erstellen des Prozessorobjekts ausgel�st: javax.annotation.processing.Processor: Provider org.neo4j.annotations.service.ServiceAnnotationProcessor could not be instantiated
2023-09-29 10:35:22,147 ERROR [io.qua.dep.dev.JavaCompilationProvider] (Timer-0) Ung�ltige Servicekonfigurationsdatei, oder Ausnahme beim Erstellen des Prozessorobjekts ausgel�st: javax.annotation.processing.Processor: Provider org.neo4j.annotations.service.ServiceAnnotationProcessor could not be instantiated, line -1 in [unknown source]
2023-09-29 10:35:24,319 ERROR [io.qua.dep.dev.JavaCompilationProvider] (Timer-0) Ung�ltige Servicekonfigurationsdatei, oder Ausnahme beim Erstellen des Prozessorobjekts ausgel�st: javax.annotation.processing.Processor: Provider org.neo4j.annotations.service.ServiceAnnotationProcessor could not be instantiated, line -1 in [unknown source]
```

Because I like to use Neo4J embedded, I'm not using the actual Quarkus Neo4J client dependency, but the official Neo4J dependencies instead, because I want to use the **embedded neo4j.**
 
```
  implementation("org.neo4j:neo4j:$neo4jVersion") 
```

and

```
    implementation("org.neo4j:neo4j-kernel:$neo4jVersion")
    implementation("org.neo4j:annotations:$neo4jVersion")
```

**Comment by klaren** [comment]
The annotations maven module contains service-loaded preprocessors, https://github.com/neo4j/neo4j/blob/5.10/annotations/src/main/resources/META-INF/services/javax.annotation.processing.Processor

I know very little about Quarkus, but it looks like the Quarkus compiler is picking this up and trying to execute them. Is there any way you can exclude them?

Disregarding that, I strongly suspect that this is not the only problem you will be facing. Neo4j is not designed to be hot-reloaded and will probably fail in mysterious ways. Neo4j is holding system resources, allocates memory outside of the JVM, etc. which is problematic in a reload scenario.

**Comment by michael-simons** [comment]
The way Quarkus does compile time optimizations in all areas makes it unlikely that neo4j embedded can be beneficially put to use , really. I doubt the effort is worth the gain. 

We do provide a proper integration for the driver and for OGM.

**Comment by mklueh** [comment]
@klaren not sure, but I assume you can do it somehow with Gradle.

Does this problem occur because the annotation processor is part of the main package, instead of being wrapped into a separate annotation processor library like Lombok or MapStruct do it?

> annotationProcessor("org.projectlombok:lombok:$lombokVersion")

But if there are presumably other potential problems I'll use the standalone Neo4J server instead :) Thank you for the clarification

**Comment by klaren** [comment]
> Does this problem occur because the annotation processor is part of the main package, instead of being wrapped into a separate annotation processor library like Lombok or MapStruct do it?

Yes. They are automatically service-loaded whenever you build something with neo4j to simplify plugin/procedure development. :smiley:

**Comment by michael-simons** [comment]
@mklueh as you stated that you are using stand alone server, we are gonna close this. If you have any issues or questions wrt the [Quarkus Neo4j integration](https://github.com/quarkiverse/quarkus-neo4j) please reach out there. Thank you for using Neo4j.
